pkg/diff/diff.go

```diff
@@ -283,6 +285,28 @@ func removeWebhookMutation(predictedLive, live *unstructured.Unstructured, gvkPa
 	return &unstructured.Unstructured{Object: pl}, nil
 }
 
+// Appends the key fields like "name" to the set elements which have some entries keyed by those fields.
+// This is needed to merge properly, see https://github.com/argoproj/argo-cd/issues/20792.
+func appendKeyFields(set *fieldpath.Set) {
```

Comment:
Is there a good way to unit test this function? I understand it's covered by higher-level function tests, but I think some tests specific to this function would really clarify how it works.

Reply:
I'll look into this

Reply:
Glad I did a test - it caught a subtle bug.